Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift

```diff
@@ -64,74 +83,174 @@ public struct LivestreamPlayer<Factory: ViewFactory>: View {
+        self.muted = muted
+        self.showParticipantCount = showParticipantCount
+        _state = ObservedObject(wrappedValue: call.state)
         self.joinPolicy = joinPolicy
         self.showsLeaveCallButton = showsLeaveCallButton
         self.onFullScreenStateChange = onFullScreenStateChange
-        viewModel.call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
+        call.updateParticipantsSorting(with: livestreamOrAudioRoomSortPreset)
     }
 
     public var body: some View {
         ZStack {
-            if viewModel.errorShown {
+            if livestreamState == .error {
                 errorView
-            } else if viewModel.loading {
+            } else if livestreamState == .joining {
                 loadingView
-            } else if state.backstage {
+            } else if livestreamState == .backstage {
                 notStartedView
             } else {
                 videoRenderer
+                livestreamControls
             }
-            livestreamControls
         }
         .onChange(of: state.participants, perform: { newValue in
-            if viewModel.muted && newValue.first?.track != nil {
-                viewModel.muteLivestreamOnJoin()
+            if muted && newValue.first?.track != nil {
+                muteLivestreamOnJoin()
             }
         })
+        .onChange(of: call.state.backstage) { _ in
+            livestreamState = call.state.backstage ? .backstage : .live
+            if let startsAt = state.startsAt, livestreamState == .backstage && timer == nil {
+                timer = Timer.scheduledTimer(withTimeInterval: 1, repeats: true) { _ in
+                    Task { @MainActor in
+                        countdown = startsAt.timeIntervalSinceNow
+                        if countdown <= 0 {
+                            stopTimer()
+                            countdown = 0
+                        }
+                    }
+                }
+            }
+            if livestreamState == .live {
+                stopTimer()
+            }
+        }
+        .onChange(of: controlsShown) { _ in
+            if controlsShown {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+                    if !self.streamPaused {
+                        self.controlsShown = false
+                    }
+                }
+            }
+        }
         .onAppear {
             switch joinPolicy {
             case .none:
                 break
             case .auto:
-                viewModel.joinLivestream()
+                joinLivestream()
             }
         }
         .onDisappear {
             switch joinPolicy {
             case .none:
                 break
             case .auto:
-                viewModel.leaveLivestream()
+                leaveLivestream()
             }
         }
     }
+    
+    func toggleAudioOutput() {
+        Task {
```

Comment:
Can we at least store them in a `DisposableBag` so we avoid any issues where the View has been removed from the hierarchy but nested tasks keeps running?

Reply:
don't think that is possible, but I will store them just in case